lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -81,17 +81,19 @@ def _is_collection_dir(dir_path):
 
 def _find_collections_in_subdirs(dir_path):
     b_dir_path = to_bytes(dir_path, errors='surrogate_or_strict')
-    galaxy_yml_glob_pattern = os.path.join(
+
+    subdir_glob_pattern = os.path.join(
         b_dir_path,
         # b'*',  # namespace is supposed to be top-level per spec
         b'*',  # collection name
-        _GALAXY_YAML,
-    )
-    return (
-        os.path.dirname(galaxy_yml)
-        for galaxy_yml in iglob(galaxy_yml_glob_pattern)
     )
 
+    for subdir in iglob(os.path.join(subdir_glob_pattern)):
+        if os.path.isfile(os.path.join(subdir, _MANIFEST_JSON)):
+            yield subdir
```

Comment:
Do we *want to* prefer manifest over galaxy.yml?

Reply:
For consistency, I think so. Otherwise a top level collection in a dir and a collection in a subdir are handled differently.

Reply:
Since this function just returns the directory I don't think it makes any difference here.